### ISSUE TITLE
gapless playback

### ISSUE THREAD

**Issue by LukeZBaker** [timdubbins/tap#2]
I'm mighty impressed with tap.

I note it seeks inspiration from cmus. That offers gapless playback. Is there any chance of adding gapless playback for tap please?

About 10% of my music collection is ruined without it.

**Comment by timdubbins** [comment]
Thanks for the feedback and glad you're enjoying the app!

Gapless playback is something I intend to implement. It's actually more of a bug that I've pushed to the back of the pile as Symphonia (which we use for decoding via Rodio) already provides it. I've been putting it off fixing it because I intended to remove the Rodio dependency at some point. 

The last update fixed a lot of things so I hope to fix this and also implement file seeking in the next update.

**Comment by LukeZBaker** [comment]
That's great to hear. I'll write up a review of tap once gapless playback has been implemented.

**Comment by timdubbins** [comment]
So I just pushed a release that should implement gapless playback (seeking will have to be for another time). If you could let me know if this is working as you'd expect that would be appreciated!

**Comment by LukeZBaker** [comment]
Yes, from limited testing, gapless playback is working as expected! Thanks. I'm going to crack on with my review.

It'll be easier to verify everything is ok when track seeking is implemented.

**Comment by timdubbins** [comment]
Yeah, especially when tracks that use it tend to be on the longer side! Unfortunately seeking requires changes that are a bit more extensive, going to have to think about the neatest way to incorporate it

Thanks for offering to write a review. It will be useful to hear any negatives so don't hold back!

**Comment by LukeZBaker** [comment]
I wrote up my review fwiw. I'm using tap as my daily music driver!

**Comment by timdubbins** [comment]
Thanks for writing such a detailed review, it's very useful to hear how someone experiences using the app. 

The suggestions you make I think are pretty good. 

I think I will add an independent volume control as I have an idea of how to add it without cluttering the UI. 

I also have code that speeds up the initial loading for a user set directory by using a cache, just need to iron out some of the wrinkles with that.

Also, pretty sure I introduced the 'occasional blinking' when I added gapless playback. Should be able to fix that.

Aside from these changes, plus seeking, I think the program will be in a nice state.

Further suggestions / bug reports are very welcome!

**Comment by timdubbins** [comment]
Just a heads up, latest release adds volume control and the ability to set a default directory for a faster startup

**Comment by LukeZBaker** [comment]
I look regularly to see if there are new commits, clone and build without waiting for an actual release as I'm that keen to see developments :)

**Comment by timdubbins** [comment]
Latest release implements seeking. Also fixes the annoying the 'blink' that occurred every once in a while between track changes

**Comment by LukeZBaker** [comment]
Awesome Tim. I'm going to publish an updated review as I was waiting for seeking to be implemented!

**Comment by timdubbins** [comment]
That would be great, thanks!

**Comment by LukeZBaker** [comment]
I've written a review of the latest version.

tap is just so cool now.

11 GitHub stars -- meh. tap deserves 11K.

**Comment by timdubbins** [comment]
Thanks for taking the time to do that. Glad you're enjoying some of the new features!

I agree a config file would be useful now we have the color option. Might look at adding that next

**Comment by LukeZBaker** [comment]
A bug in v0.4.10 is evident (I did all my testing using v0.4.9 as v0.4.10 was released just after finishing my review). 

I've noticed that when v0.4.10 finishes playing a track, it just repeats that track (though the interface indicates the next track is being played). This happens with the precompiled binary and the deb package.

**Comment by timdubbins** [comment]
Thanks Luke. 

I cleaned some of the cruft in the last release, as well as the line of code that increments the file in the queue...🤦‍♂️

**Comment by timdubbins** [comment]
This should be fixed now!

**Comment by LukeZBaker** [comment]
Yes, lovely jubbly!
